### ISSUE TITLE
Requested: http-cookie-agent update to V6.0.6 (reason: error: Module not found: Can't resolve 'deasync' in nextjs (turbopack))

### ISSUE THREAD

**Issue by mniesen** [ewsjs/xhr#23]
Hello

After using @ewsjs/xhr v3.1.3 and ews-javascript-api v0.15.3 in nextjs v15.0.1 with turbopack, I got the following error:
```
./node_modules/@ewsjs/xhr/node_modules/http-cookie-agent/dist/utils/save_cookies_from_header.js:18:3
Module not found: Can't resolve 'deasync'
  16 |   const setCookieSync = async_UNSTABLE ?
  17 |   // eslint-disable-next-line @typescript-eslint/no-var-requires
> 18 |   require('deasync')(jar.setCookie.bind(jar)) : jar.setCookieSync.bind(jar);
     |   ^^^^^^^^^^^^^^^^^^
  19 |   for (const cookie of [cookies].flat()) {
  20 |     if (cookie == null) {
  21 |       continue;



https://nextjs.org/docs/messages/module-not-found
```

After installing deasync manually, the error changed to:
```
 ⨯ ./node_modules/deasync
Module not found
server relative imports are not implemented yet. Please try an import relative to the file you are importing from.


https://nextjs.org/docs/messages/module-not-found
```

Note: Without using turbopack it don't get the error.

The error happens in the http-cookie-agent package, that is installed under @ewsjs/xhr/node_modules. The version of this package is "^5.0.4". I searched in their code on Github and the latest http-cookie-agent version (v6.0.6) doesn't use deasync anymore. I uninstalled the current http-cookie-agent and installed the latest version (v6.0.6) under @ewsjs/xhr/node_modules, so the @ewsjs/xhr package now uses this version. This update resolved the error.

<br/>
<br/>

**My request**: Can you please update the http-cookie-agent package in @ews/xhr to V6.0.6 so the deasync error is resolved?

Kind regards,
Michiel Niesen

**Comment by gautamsi** [comment]
are you able to override the package using yarn resolutions or pnpm overrides? does that resolve the issue?

**Comment by mniesen** [comment]
Hello

Thanks for your quick response.
I searched online on how to override the package version of http-cookie-agent with yarn or pnpm, because I didn't know that that was possible. I found that there is a builtin feature in npm.

After adding:
```
"overrides": {
    "ews-javascript-api": {
      "http-cookie-agent": "6.0.6"
    }
  }
```
to the package.json file and then installing ews-javascript-api, the error is resolved. After looking in the node_modules folder, I can confirm that v6.0.6 is now being used.

Kind regards,
Michiel Niesen
